File: 351004/Brazhalovich/Lab_1/discussion/src/main/java/org/example/discussion/repository/CommentRepository.java
package org.example.discussion.repository;

import org.example.discussion.entity.Comment;
import org.springframework.data.cassandra.repository.CassandraRepository;
import org.springframework.data.cassandra.repository.Query;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;

@Repository
public interface CommentRepository extends CassandraRepository<Comment, Long> {

    @Query("SELECT * FROM tbl_comment WHERE news_id = ?0 ALLOW FILTERING")
    List<Comment> findByNewsId(Long newsId);

    Optional<Comment> findById(Long id);
}

File: 351003/Soika/entities-app/discussion/src/main/kotlin/com/example/entitiesapp/repository/CommentRepository.kt
package com.example.entitiesapp.repository

import com.example.entitiesapp.model.Comment
import org.springframework.data.cassandra.repository.CassandraRepository
import org.springframework.data.cassandra.repository.Query
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface CommentRepository : CassandraRepository<Comment, Long> {

    @Query("SELECT * FROM tbl_comment WHERE id = ?0")
    fun findByCommentId(id: Long): Optional<Comment>

    @Query("DELETE FROM tbl_comment WHERE story_id = ?0")
    fun deleteAllByStoryId(storyId: Long)

    @Query("DELETE FROM tbl_comment WHERE story_id = ?0 AND id = ?1")
    fun deleteByCompositeKey(storyId: Long, id: Long)
}